server/src/com/cloud/api/ApiResponseHelper.java

```diff
@@ -526,16 +529,18 @@ public static DataStoreRole getDataStoreRole(Snapshot snapshot, SnapshotDataStor
         }
 
         long storagePoolId = snapshotStore.getDataStoreId();
-        DataStore dataStore = dataStoreMgr.getDataStore(storagePoolId, DataStoreRole.Primary);
+        if (snapshotStore.getState() != null && ! snapshotStore.getState().equals(ObjectInDataStoreStateMachine.State.Destroyed)) {
+            DataStore dataStore = dataStoreMgr.getDataStore(storagePoolId, DataStoreRole.Primary);
 
-        Map<String, String> mapCapabilities = dataStore.getDriver().getCapabilities();
+            Map<String, String> mapCapabilities = dataStore.getDriver().getCapabilities();
 
-        if (mapCapabilities != null) {
-            String value = mapCapabilities.get(DataStoreCapabilities.STORAGE_SYSTEM_SNAPSHOT.toString());
-            Boolean supportsStorageSystemSnapshots = new Boolean(value);
+            if (mapCapabilities != null) {
+                String value = mapCapabilities.get(DataStoreCapabilities.STORAGE_SYSTEM_SNAPSHOT.toString());
+                Boolean supportsStorageSystemSnapshots = new Boolean(value);
```

Comment:
`new Boolean` skips the constant pool -- putting unnecessary pressure on the heap and creating a potential memory leak.  Please use `Boolean.valueOf` to part the value to avoid this issue.